test/cppsuite/test_harness/workload/random_generator.cxx

```diff
@@ -37,25 +42,37 @@ random_generator::instance()
 }
 
 std::string
-random_generator::generate_string(std::size_t length)
+random_generator::generate_string(std::size_t length, characters_type type)
 {
-    std::string random_string;
+    std::string str;
 
-    for (std::size_t i = 0; i < length; ++i)
-        random_string += _characters[_distribution(_generator)];
+    switch (type) {
+    case characters_type::ALPHABET:
+        str = _alphabet;
+        break;
+    case characters_type::PSEUDO_ALPHANUMERIC:
+        str = _pseudo_alphanum;
+        break;
+    default:
+        testutil_die(type, "Unexpected characters_type");
+        break;
+    }
 
-    return (random_string);
+    std::shuffle(str.begin(), str.end(), _generator);
+    return (str.substr(0, length));
 }
 
 std::string
-random_generator::generate_pseudo_random_string(std::size_t length)
+random_generator::generate_pseudo_random_string(std::size_t length, characters_type type)
 {
     std::string random_string;
-    std::size_t start_location = _distribution(_generator);
+    std::uniform_int_distribution<> &distribution = get_distribution(type);
+    std::size_t start_location = distribution(_generator);
+    const std::string &characters = get_characters(type);
 
     for (std::size_t i = 0; i < length; ++i) {
```

Comment:
For pseudo random string generation, how I understand it is that we find a random start location and for each character we increment the the start location. Is this what we want? e.g. HIJKL or 12345

Reply:
Yes, that's correct. I believe it was for a better compression according to @luke-pearson's comment:
`/* Generate a pseudo random string which compresses better. */`

Reply:
Truely pseudo-random strings would not compress well, and so I think it would be useful to ensure clearly we document the types of pseudo-random strings that can be generated.